File: backend/byteBlog/src/main/java/com/jeswin8801/byteBlog/entities/dto/blog/AuthorDto.java
package com.jeswin8801.byteBlog.entities.dto.blog;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.jeswin8801.byteBlog.entities.model.Role;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Set;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AuthorDto {

    private String username;

    @JsonProperty("full-name")
    private String fullName;

    private String email;

    @JsonProperty("profile-image-url")
    private String profileImageUrl;

    private String about;
}

File: frontend/src/app/models/dtos/blog/author-dto.ts
import { JsonProperty } from 'json-object-mapper';

export class AuthorDto {
  email?: string;

  username?: string;

  @JsonProperty({ name: 'full-name' })
  fullName?: string;

  about?: string;

  @JsonProperty({ name: 'profile-image-url' })
  profileImageUrl?: string;
}
